core/src/main/scala/org/scalatra/ScalatraBase.scala

```diff
@@ -544,23 +546,23 @@ trait ScalatraBase
     case _ => response.status.code
   }
 
-  def get(transformers: RouteTransformer*)(action: => Any): Route = addRoute(Get, transformers, action)
+  def get(transformers: RouteTransformer*)(action: => Any): Route = macro RouteMacros.getImpl
 
-  def post(transformers: RouteTransformer*)(action: => Any): Route = addRoute(Post, transformers, action)
+  def post(transformers: RouteTransformer*)(action: => Any): Route = macro RouteMacros.postImpl
 
-  def put(transformers: RouteTransformer*)(action: => Any): Route = addRoute(Put, transformers, action)
+  def put(transformers: RouteTransformer*)(action: => Any): Route = macro RouteMacros.putImpl
 
-  def delete(transformers: RouteTransformer*)(action: => Any): Route = addRoute(Delete, transformers, action)
+  def delete(transformers: RouteTransformer*)(action: => Any): Route = macro RouteMacros.deleteImpl
 
-  def trap(codes: Range)(block: => Any): Unit = {
-    addStatusRoute(codes, block)
-  }
+  def options(transformers: RouteTransformer*)(action: => Any): Route = macro RouteMacros.optionsImpl
+
+  def head(transformers: RouteTransformer*)(action: => Any): Route = macro RouteMacros.headImpl
 
-  def options(transformers: RouteTransformer*)(action: => Any): Route = addRoute(Options, transformers, action)
+  def patch(transformers: RouteTransformer*)(action: => Any): Route = macro RouteMacros.patchImpl
 
-  def head(transformers: RouteTransformer*)(action: => Any): Route = addRoute(Head, transformers, action)
+  def trap(codes: Range)(block: => Any): Unit = macro RouteMacros.trapImpl
 
-  def patch(transformers: RouteTransformer*)(action: => Any): Route = addRoute(Patch, transformers, action)
+  implicit def int2range(x: Int) = Range(x, x + 1)
```

Comment:
Is it really necessary? In this case, just defining two methods would be cleaner, IMO.